src/utils/read-file.ts

```diff
@@ -1,95 +1,59 @@
 import R from 'ramda'
-import fs from 'fs'
-import glob from 'glob'
-import yaml from 'js-yaml'
+import path from 'path'
+
+import { findConfigFiles, findFiles } from './find-files'
+import { getFileFromPath } from './get-file-from-path'
 
 export enum FileExtension {
   JSON = 'json',
   YAML = 'yaml',
   YML = 'yml',
 }
-type File = {
+export type File = {
   contents: EncryptedConfig | Schema
   filePath: string
 }
 
-export class NoFileError extends Error {
-  constructor(message: string) {
-    super(message)
-    Object.setPrototypeOf(this, NoFileError.prototype)
-  }
-}
-
-const getFileExtensionPattern = (): string =>
+export const getFileExtensionPattern = (): string =>
   `{${Object.values(FileExtension).join(',')}}`
 
-const findConfigFiles = (
-  basePath: string,
-  filenameWithoutExtension: string
-): string[] => {
-  const globPattern = `${basePath}/${filenameWithoutExtension}.${getFileExtensionPattern()}`
-
-  return glob.sync(globPattern, {})
-}
-
-const getFileExtensionFromPath = R.compose<
-  string,
-  string[],
-  string,
-  FileExtension
->(
-  ext => FileExtension[ext.toUpperCase() as keyof typeof FileExtension],
-  R.last,
-  R.split('.')
-)
-
-const parseConfig = (configFile: string): File => {
-  const fileExtension = getFileExtensionFromPath(configFile)
-  const fileContent = fs.readFileSync(configFile).toString()
-
-  return {
-    contents:
-      fileExtension === FileExtension.JSON
-        ? JSON.parse(fileContent)
-        : yaml.load(fileContent),
-    filePath: configFile,
-  }
-}
-
-export const readConfigFile = (filename: string | undefined): File => {
-  if (R.isNil(filename)) {
-    throw new Error('Config file name must not be nil')
-  }
-
-  const path = `${process.cwd()}/config`
-  const files = findConfigFiles(path, filename)
+export const readConfigFile = (basePath: string, fileName: string): File => {
+  const filePaths = findConfigFiles(basePath, fileName)
 
-  if (!R.is(Array, files) || R.isEmpty(files)) {
-    throw new NoFileError(
-      `None of ${filename}.${getFileExtensionPattern()} found. One of these files must exist.`
+  if (R.isEmpty(filePaths)) {
+    throw new Error(
+      `None of ${fileName}.${getFileExtensionPattern()} found. One of these files must exist.`
     )
-  } else if (files.length > 1) {
+  } else if (filePaths.length > 1) {
     throw new Error(
-      `More than one of ${filename}.${getFileExtensionPattern()} found. Exactly one must exist.`
+      `More than one of ${fileName}.${getFileExtensionPattern()} found. Exactly one must exist.`
```

Comment:
not sure i'm reading this correctly. but there can never be two `dev.yaml` files in the same directory in most common file systems

Reply:
but there can be a `dev.yaml` and `dev.yml`, `dev.json` in the same dir, which this check would correctly prohibit. Otherwise, it would mean there are multiple config files for one environment, which isn't supported.

Reply:
got it, thx

Reply:
got it 👍